### PR TITLE
[RFC] Remove 'ttyfast'

### PR DESCRIPTION
**Pull request by ghost**
refs #1045 #1051

`ttyfast` was enabled by default a while ago, and has apparently not created any issues (at least noted on GH). The amount of actual code related to it was tiny, so I just removed it.

### CODE REVIEW

**Comment by tarruda** [comment]
:+1:

File: runtime/doc/term.txt [hunk 1]
```diff
@@ -518,50 +517,7 @@ One command can be used to set the screen size:
...
-lines jump up and down.  If you don't want this set the 'ttyfast' option.
-This will redraw the window instead of scroll it.
-
-If your terminal scrolls very slowly, but redrawing is not slow, set the
-'ttyscroll' option to a small number, e.g., 3.  This will make Vim redraw the
-screen instead of scrolling, when there are more than 3 lines to be scrolled.
-
-If you are using a color terminal that is slow, use this command: >
-	hi NonText cterm=NONE ctermfg=NONE
-This avoids that spaces are sent when they have different attributes.  On most
```
Comment (justinmk) [inline runtime/doc/term.txt]: This tip is still useful, and others.

Comment (Pyrohh) [inline runtime/doc/term.txt]: Oops, I didn't mean to push that.

**Comment by Pyrohh** [comment]
Squashed and repushed with the recommended changes.

**Comment by Pyrohh** [comment]
I left a comment in `screen.c` for a condition I was unsure about (I'm pretty sure comments on my branch aren't sent to people subscribed to this PR).

**Comment by justinmk** [comment]
My first instinct wonders if this is a feature that could still be useful. But if I understand it correctly, this is only useful on terminals that do not support a "scrolling region". After searching, I have the impression that most terminals support scrolling regions. So the presence of `'ttyfast'` may do more harm than good for users, insofar as it adds a useless knob for users to waste time investigating.

So LGTM.

**Comment by justinmk** [comment]
@tarruda does the `'ttybuiltin'` option make sense for Neovim to support after #1820?

**Comment by justinmk** [comment]
merged

**Comment by tarruda** [comment]
> @tarruda does the 'ttybuiltin' option make sense for Neovim to support after #1820?

No, it doesn't. After #1820 I will create a PR to remove remaining dead code

**Pull request closed without merging**